### PR TITLE
ruby@3.0 3.0.7 and deprecate

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

- [x] `resource` blocks have been checked for updates.

### CODE REVIEW

**Comment by chenrui333** [comment]
to avoid merge conflict, will do autobump as followup

**Comment by cho-m** [comment]
> to avoid merge conflict, will do autobump as followup

It's EOL (3.0.7 as final release) so should deprecate rather than autobump - https://www.ruby-lang.org/en/downloads/branches/
```
status: eol
release date: 2020-12-25
normal maintenance until: 2023-04-01
EOL: 2024-04-23
```

**Comment by chenrui333** [comment]
> > to avoid merge conflict, will do autobump as followup
> 
> It's EOL (3.0.7 as final release) so should deprecate rather than autobump - https://www.ruby-lang.org/en/downloads/branches/
> 
> ```
> status: eol
> release date: 2020-12-25
> normal maintenance until: 2023-04-01
> EOL: 2024-04-23
> ```

will do the deprecate along with this bump

File: Formula/r/ruby@3.0.rb [hunk 1]
```diff
@@ -23,6 +17,8 @@ class RubyAT30 < Formula
     sha256 x86_64_linux:   "d04b41e0aeb4155e22d69dc8fabc8d6fbf09703476b552b3de2de88322ec0654"
   end
 
+  deprecate! date: "2024-04-24", because: unmaintained
```
Comment (SMillerDev) [inline Formula/r/ruby@3.0.rb:20]: Let's disable it in a year then (and deprecate 3.1)

File: Formula/r/ruby@3.0.rb [hunk 2]
```diff
@@ -25,6 +19,8 @@ class RubyAT30 < Formula
 
   keg_only :versioned_formula
 
+  disable! date: "2025-04-23", because: unmaintained
```
[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/r/ruby@3.0.rb]: 👍

**Review by p-linnane — APPROVED** [review]

File: Formula/r/ruby@3.0.rb [hunk 3]
```diff
@@ -37,14 +33,8 @@ class RubyAT30 < Formula
   # The exception is Rubygem security fixes, which mandate updating this
   # formula & the versioned equivalents and bumping the revisions.
   resource "rubygems" do
-    url "https://rubygems.org/rubygems/rubygems-3.4.10.tgz"
-    sha256 "55f1c67fa2ae96c9751b81afad5c0f2b3792c5b19cbba6d54d8df9fd821460d3"
-  end
-
-  # Update the bundled openssl gem for compatibility with OpenSSL 3.
-  resource "openssl" do
-    url "https://github.com/ruby/openssl/archive/refs/tags/v3.1.0.tar.gz"
```
Comment (cho-m) [inline Formula/r/ruby@3.0.rb:46]: Probably still need this. v2.2.2 is bundled in Ruby 3.0.7.

**Review by p-linnane — APPROVED** [review]

**Review by cho-m — APPROVED** [review]

**Pull request merged**